examples/01_basic_ping_bot/src/main.rs

```diff
@@ -59,54 +58,74 @@ impl From<TungsteniteError> for Error {
 }
 
 fn main() {
-    tokio::run(try_main().map_err(|why| {
-        println!("Error running shard: {:?}", why);
-    }).boxed().compat(TokioDefaultSpawner));
+    tokio::run(main_future().map_err(|reason| {
+        eprintln!("Error when running shard {:?}:", reason);
+    }).boxed().compat());
 }
 
-async fn try_main() -> Result<(), Error> {
+async fn main_future() -> Result<(), Error> {
```

Comment:
No reason to change the name of the function, and `try_main` is more idiomatic for the name of the "main" function returning a `Result` enclosed by the true `main`.